doc/lua_api.md

```diff
@@ -1108,6 +1109,17 @@ Table used to specify how a sound is played:
     -- Attach the sound to an object.
     -- Can't be used together with `pos`.
 
+    -- For backwards compatibility, sounds continue playing
+    -- at the last location of the object if an object is removed
+    -- (for example if an entity dies).
+
+    -- It is not recommended to rely on this.
+    -- For death sounds, prefer playing a positional sound at the objects last location.
+
+    -- Sounds stop playing on clients if objects leave the active object range;
+    -- they should start playing again if objects come back into range
+    -- (but due to a known bug, they don't yet).
```

Comment:
Missing pieces of information:
- explicit recommendation to call minetest.sound_stop in on_deactivate if you want the sound to stop
- different behavior of ephemeral sounds